File: 2025/1429. First Unique Number/Solution.java
class FirstUnique {

    Map<Integer, Integer> map = new HashMap<>();
    Queue<Integer> queue = new LinkedList<>();

    public FirstUnique(int[] nums) {
        for (int n: nums) {
            add(n);
        }
    }
    
    public int showFirstUnique() {
        while (!queue.isEmpty()) {
            if (map.get(queue.peek()) == 1) {
                return queue.peek();
            }
            queue.remove();
        }
        return -1;
    }
    
    public void add(int value) {
        if (map.containsKey(value)) {
            map.put(value, map.get(value) + 1);
        } else {
            map.put(value, 1);
            queue.add(value);
        }
    }
}

/**
 * Your FirstUnique object will be instantiated and called as such:
 * FirstUnique obj = new FirstUnique(nums);
 * int param_1 = obj.showFirstUnique();
 * obj.add(value);
 */

File: 2025/1429. First Unique Number/solution.py
class FirstUnique:

    def __init__(self, nums: List[int]):
        self.num_map = {}
        self.queue = deque()
        for n in nums:
            self.add(n)


    def showFirstUnique(self) -> int:
        while self.queue:
            if (self.num_map[self.queue[0]] == 1):
                return self.queue[0]
            self.queue.popleft()
        return -1

    def add(self, value: int) -> None:
        if value in self.num_map:
            self.num_map[value] += 1
        else:
            self.num_map.update({value: 1})
            self.queue.append(value)




# Your FirstUnique object will be instantiated and called as such:
# obj = FirstUnique(nums)
# param_1 = obj.showFirstUnique()
# obj.add(value)